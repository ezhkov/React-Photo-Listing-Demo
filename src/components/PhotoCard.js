import React, { Component } from 'react';
import { Button, Card, CardBody, CardImage, CardTitle, CardText } from 'mdbreact';


export class PhotoCard extends Component {

    state = {};


    render() {
        return (
            <Card>
                <CardImage className="img-fluid" src={this.props.photoName} />
                <CardBody>
                    <CardTitle>Card title</CardTitle>
                    <CardText>Some quick example text to build on the card title and make up the bulk of the card's content.</CardText>
                    <Button href="#">Button</Button>
                </CardBody>
            </Card>
        );
    }

}