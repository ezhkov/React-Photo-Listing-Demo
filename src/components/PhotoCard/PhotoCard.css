.badge {
    margin-right: 5px;
}
.photo-list {
    max-width: 900px;
    margin: 0 auto;
}
div.photo-card-self .photo-card-self__link-ava {
    display: flex;
    justify-content: center;
    margin-bottom: 10px;
}
div.photo-card-self .photo-card-self__badge {
    margin: 5px 3px;
    cursor: pointer;
}
div.photo-card-self .photo-card__photo-link {
    position: relative;
    padding-bottom: 75%;
}

div.photo-card-self .photo-card__img {
    position: absolute;
    height: 100%;
    max-width: none;
    left: 50%;
    transform: translate(-50%,0);
    max-width: 100%;
}