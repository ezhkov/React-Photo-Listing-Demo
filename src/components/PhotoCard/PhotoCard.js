import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  Card,
  CardBody,
  CardImage,
  CardTitle,
  CardText,
} from 'mdbreact';

const PhotoCard = ({ photoName }) => (
  <Card>
    <CardImage className="img-fluid" src={photoName} />
    <CardBody>
      <CardTitle>Card title</CardTitle>
      <CardText>
        Some quick example text to build on the card title
        and make up the bulk of the card&apos;s content.
      </CardText>
      <Button href="#">Button</Button>
    </CardBody>
  </Card>);


PhotoCard.propTypes = {
  photoName: PropTypes.string,
};
PhotoCard.defaultProps = {
  photoName: '',

};

export default PhotoCard;
