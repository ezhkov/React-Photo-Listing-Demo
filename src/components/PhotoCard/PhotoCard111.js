// import React from 'react';
// import { LastLocationProvider } from 'react-router-last-location';
// import { BrowserRouter } from 'react-router-dom';
// import Enzyme, { shallow, mount } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
// import PhotoCard from '.';

// Enzyme.configure({ adapter: new Adapter() });

// describe('Test of component of PhotoCard', () => {
//   it('Test state PhotoCard dive props', () => {
//     const props = {
//       photoName: '',
//     };
//     const card = shallow(
//       <BrowserRouter>
//         <LastLocationProvider><PhotoCard {...props} /></LastLocationProvider>
//       </BrowserRouter>,
//     );
//     expect(typeof card.props().children.props.children.props.photoName).toEqual('string');
//   });

//   it('Test state PhotoCard props', () => {
//     const props = {
//       photoName: 'src',
//       photoID: '1',
//       userID: '1',
//     };
//     const card = mount(
//       <BrowserRouter>
//         <LastLocationProvider><PhotoCard {...props} /></LastLocationProvider>
//       </BrowserRouter>,
//     );
//     expect(card.find('.photo-card__img').prop('src')).toEqual('src');
//     expect(card.find('a.photo-card__photo-link').prop('href')).toEqual('/photo/1');
//     expect(card.find('a.photo-card-self__link-ava').prop('href')).toEqual('/users/1');
//   });

//   it('Test state PhotoCard booleans', () => {
//     const props1 = {
//       title: '11',
//       photoDesc: '11',
//       tags: [
//         {
//           title: '1',
//         },
//         {
//           title: '2',
//         },
//         {
//           title: '3',
//         },
//         {
//           title: '4',
//         },
//       ],
//     };

//     const card1 = mount(
//       <BrowserRouter>
//         <LastLocationProvider><PhotoCard {...props1} /></LastLocationProvider>
//       </BrowserRouter>,
//     );
//     expect(card1.find('a.photo-card-self__link-ava').length).toEqual(1);
//     expect(card1.find('.photo-card-self__desc').text()).toEqual('11');
//     expect(card1.find('Popover').length).toEqual(1);
//     const props2 = {
//       title: '',
//       photoDesc: '',
//       tags: [
//         {
//           title: '1',
//         },
//         {
//           title: '2',
//         },
//         {
//           title: '3',
//         },
//       ],
//     };

//     const card2 = mount(
//       <BrowserRouter>
//         <LastLocationProvider><PhotoCard {...props2} /></LastLocationProvider>
//       </BrowserRouter>,
//     );
//     expect(card2.find('a.photo-card-self__link-ava').length).toEqual(0);
//     expect(card2.find('.photo-card-self__desc').text()).toEqual('No Description');
//     expect(card2.find('Popover').length).toEqual(0);
//   });
//   it('Test snapshot PhotoCard component', () => {
//     const card = shallow(
//       <BrowserRouter>
//         <LastLocationProvider><PhotoCard /></LastLocationProvider>
//       </BrowserRouter>,
//     );
//     expect(card).toMatchSnapshot();
//   });
// });