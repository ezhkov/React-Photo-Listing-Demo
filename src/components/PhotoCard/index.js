import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Card, Avatar, Tag, Popover } from "antd";
import './PhotoCard.css';
const { Meta } = Card;

class PhotoCard extends Component {
  constructor(...args) {
    super(...args);
    this.state = {
      visible: false,
    };
  }

  showModal = () => {
    this.setState({
      visible: true,
    });
  }

  hideModal = () => {
    this.setState({
      visible: false,
    });
  }

  render() {
    const {
      tags,
      photoName, 
      photoDesc, 
      title, 
      photoID, 
      userID, 
      userAvatar, 
      onSearchTagValue,
    } = this.props;
    return (
      <div className="photo-card-self">
          <Card
            style={{ width: '100%' }}
            cover={
              <Link className="photo-card__photo-link" to={`/photo/${photoID}`}>
                <img
                  className="photo-card__img"
                  alt="example"
                  src={photoName}
                />
              </Link>
            }
          >
        
        { title && (<Link className="photo-card-self__link-ava" to={`/users/${userID}`}>
            <Meta
              avatar={
                <Avatar src={userAvatar} />
              }
              title={title}
            />

          </Link>)}
          <p className="photo-card-self__desc">
            {`${photoDesc ? photoDesc : 'No Description'}`}
          </p>
          <div className="photo-card-self__badge-wrap">
            {tags.map((item, i) => {
                if(i < 3) {
                  return <Tag key={item.title} onClick={() => onSearchTagValue(item.title, 'tags')} className="photo-card-self__badge" >
                      {item.title}
                    </Tag>
                  }else {
                    return null;
                  }
                } 
            )}
            
            {tags.length > 3 &&  <Popover 
                                    placement="top" 
                                    title={"Remaining tags"} 
                                    content={
                                      <div>
                                          {tags.map((item, i) => {
                                            if(i > 2) {
                                              return <Tag key={item.title} onClick={() => onSearchTagValue(item.title, 'tags')} className="photo-card-self__badge">
                                                  {item.title}
                                                </Tag>
                                              }else {
                                                return null;
                                              }
                                            } 
                                        )}
                                      </div>
                                    } 
                                    trigger="click">
                                      <Tag>more tags...</Tag>
                                  </Popover>}
          </div>
        </Card>
      </div>
    );
  }
}
  

PhotoCard.propTypes = {
  photoName: PropTypes.string,
  photoDesc: PropTypes.string,
  title: PropTypes.string,
  tags: PropTypes.arrayOf(PropTypes.object),
  userAvatar: PropTypes.string,
  onSearchTagValue: PropTypes.func,
  photoID: PropTypes.string,
  userID: PropTypes.string,
};
PhotoCard.defaultProps = {
  photoName: '',
  photoDesc: '',
  title: 'Noname',
  tags: [], 
  userAvatar: '',
  photoID: '',
  userID: '',
  onSearchTagValue: () => {},
};

export default PhotoCard;
