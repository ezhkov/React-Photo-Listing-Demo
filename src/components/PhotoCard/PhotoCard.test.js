import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PhotoCard from './PhotoCard';

Enzyme.configure({ adapter: new Adapter() });

describe('Test of component of PhotoCard', () => {
  it('Test state PhotoCard component', () => {
    const card = shallow(<PhotoCard />)
    expect(Array.isArray(card.state().tagsColor)).toEqual(true);
    expect(Array.isArray(card.state().tags)).toEqual(true);
    expect(typeof card.instance().props.photoName).toEqual('string');
    expect(typeof card.instance().props.title).toEqual('string');
  });

  it('Test snapshot PhotoCard component', () => {
    const card = shallow(<PhotoCard />);
    expect(card).toMatchSnapshot();

  });
});