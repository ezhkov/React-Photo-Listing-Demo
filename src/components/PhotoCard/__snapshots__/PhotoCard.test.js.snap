// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of component of PhotoCard Test snapshot PhotoCard component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrowserRouter>
    <withRouter(LastLocationProvider)>
      <PhotoCard
        onSearchTagValue={[Function]}
        photoDesc=""
        photoID=""
        photoName=""
        tags={Array []}
        title="Noname"
        userAvatar=""
        userID=""
      />
    </withRouter(LastLocationProvider)>
  </BrowserRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <withRouter(LastLocationProvider)>
        <PhotoCard
          onSearchTagValue={[Function]}
          photoDesc=""
          photoID=""
          photoName=""
          tags={Array []}
          title="Noname"
          userAvatar=""
          userID=""
        />
      </withRouter(LastLocationProvider)>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <PhotoCard
          onSearchTagValue={[Function]}
          photoDesc=""
          photoID=""
          photoName=""
          tags={Array []}
          title="Noname"
          userAvatar=""
          userID=""
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onSearchTagValue": [Function],
          "photoDesc": "",
          "photoID": "",
          "photoName": "",
          "tags": Array [],
          "title": "Noname",
          "userAvatar": "",
          "userID": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <withRouter(LastLocationProvider)>
          <PhotoCard
            onSearchTagValue={[Function]}
            photoDesc=""
            photoID=""
            photoName=""
            tags={Array []}
            title="Noname"
            userAvatar=""
            userID=""
          />
        </withRouter(LastLocationProvider)>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <PhotoCard
            onSearchTagValue={[Function]}
            photoDesc=""
            photoID=""
            photoName=""
            tags={Array []}
            title="Noname"
            userAvatar=""
            userID=""
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onSearchTagValue": [Function],
            "photoDesc": "",
            "photoID": "",
            "photoName": "",
            "tags": Array [],
            "title": "Noname",
            "userAvatar": "",
            "userID": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
